src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,39 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string                    $key      The parameter key
+     * @param string                    $format   The expected date format
+     * @param \DateTimeInterface|string $default  The default value to be converted to a DateTime object if the parameter key does not exist
+     * @param \DateTimeZone|null        $timeZone A DateTimeZone object representing the desired time zone
+     *
+     * @return \DateTimeInterface|null
+     */
+    public function getDate($key, $format, $default = null, \DateTimeZone $timeZone = null)
+    {
+        $time = $this->get($key, $default);
+
+        if ($time instanceof \DateTimeInterface) {
+            return $default;
+        }
+
+        // if the user has specified a timezone then pass that
+        // otherwise do not even attempt to put a value but rather let the runtime decide
+        // the default value by itself
+        // this is in order to ensure compatibility with all php versions since
+        // some accept null as a TimeZone parameter and others do not
+        if (null !== $timeZone) {
+            $result = \DateTime::createFromFormat($format, $time, $timeZone);
+        } else {
+            $result = \DateTime::createFromFormat($format, $time);
+        }
+
+        // Failure to parse the date according to the specified format will return null
+        return false === $result ? null : $result;
```

Comment:
Just read @nicolas-grekas comment once more
> returning null in case of parse failure looks fine to me: this is user input, ie potentially garbage

I guess your right. Letting the app error 500 based on user input is not desired. Maybe it should only throw `InvalidArgumentException` when `$default` as a string is invalid.. wdyt?